hawkular-alerts-engine/src/main/java/org/hawkular/alerts/engine/impl/AlertsContext.java

```diff
@@ -69,4 +129,58 @@ public void registerActionListener(ActionListener listener) {
     public List<ActionListener> getActionsListeners() {
         return actionsListeners;
     }
+
+    public void notifyListeners(Set<DefinitionsEvent> notifications) {
+        Set<DefinitionsEvent.Type> notificationTypes = notifications.stream()
+                .map(n -> n.getType())
+                .collect(Collectors.toSet());
+        log.debugf("Notifying applicable listeners %s of events %s", definitionListeners, notifications);
+        definitionListeners.entrySet().stream()
+                .filter(e -> shouldNotify(e.getValue(), notificationTypes))
+                .forEach(e -> {
+                    log.debugf("Notified Listener %s of %s", e.getKey(), notificationTypes);
+                    e.getKey().onChange(notifications.stream()
+                            .filter(de -> e.getValue().contains(de.getType()))
+                            .collect(Collectors.toSet()));
+                });
+        if (!distributed) {
+            distributedListener.stream().forEach(listener -> listener.onChange(mapDistributedEvents(notifications)));
+        }
+    }
+
+    private boolean shouldNotify(Set<DefinitionsEvent.Type> listenerTypes, Set<DefinitionsEvent.Type> eventTypes) {
+        HashSet<Type> intersection = new HashSet<>(listenerTypes);
+        intersection.retainAll(eventTypes);
+        return !intersection.isEmpty();
+    }
+
+    private Set<DistributedEvent> mapDistributedEvents(Set<DefinitionsEvent> notification) {
+        if (notification == null) {
+            return null;
+        }
+        Set<DistributedEvent> events = new HashSet<>();
+        for (DefinitionsEvent definitionsEvent : notification) {
+            DistributedEvent.Operation op = null;
+            String tenantId = definitionsEvent.getTargetTenantId();
+            String triggerId = definitionsEvent.getTargetId();
+            switch (definitionsEvent.getType()) {
+                case TRIGGER_CREATE:
+                    op = DistributedEvent.Operation.ADD;
+                    break;
+                case TRIGGER_UPDATE:
+                    op = DistributedEvent.Operation.UPDATE;
+                    break;
+                case TRIGGER_REMOVE:
+                    op = DistributedEvent.Operation.REMOVE;
+                    break;
+                case TRIGGER_CONDITION_CHANGE:
+                    op = DistributedEvent.Operation.UPDATE;
+                    break;
+            }
+            if (op != null) {
+                events.add(new DistributedEvent(op, tenantId, triggerId));
+            }
+        }
+        return events;
+    }
```

Comment:
As commented, they are not really the same concept.
DefinitionsEvent has an extended target (actions) and granularity (conditions, dampening) meanwhile DistributedEvent is focus only on triggers and whole granularity as it is designed to load/unload mostly (in the core engine or alerters).